octygon-button-styler.php

```diff
@@ -1,146 +1,171 @@
-				'oct_btn_slide_none'   		=> __( 'None' ),
-                'oct_btn_slide_left'  		=> __( 'Slide Left' ),
-                'oct_btn_slide_right'  		=> __( 'Slide Right' ),
-                'oct_btn_slide_top'  		=> __( 'Slide Top' ),
-                'oct_btn_slide_bottom'  	=> __( 'Slide Bottom' ),
-                'oct_btn_slide_out_h'  		=> __( 'Slide Out Horizontal' ),
-                'oct_btn_slide_out_v'  		=> __( 'Slide Vertical' )
+			'priority' => 20,
+			'section'  => 'et_divi_buttons_hover',
+			'settings' => 'oct_btn_slide_styles',
+			'label'    => __( 'Button Slide Styles', 'oct' ),
+			'type'     => 'radio',
+			'choices'  => array(
+				'oct_btn_slide_none'   => __( 'None' ),
+				'oct_btn_slide_left'   => __( 'Slide Left' ),
+				'oct_btn_slide_right'  => __( 'Slide Right' ),
+				'oct_btn_slide_top'    => __( 'Slide Top' ),
+				'oct_btn_slide_bottom' => __( 'Slide Bottom' ),
+				'oct_btn_slide_out_h'  => __( 'Slide Out Horizontal' ),
+				'oct_btn_slide_out_v'  => __( 'Slide Vertical' ),
 			),
-		) 
+		)
 	);
 }
 add_action( 'customize_register', 'oct_customize_register' );
 
-//Octygon Customizer Javascript & CSS Output
-function oct_customize_output()
-{
-	$oct_btn_grow = ( get_option( 'oct_btn_grow_styles' ) == 'oct_btn_grow_none' ) ? '' : get_option( 'oct_btn_grow_styles' );
-	$oct_btn_shrink = ( get_option( 'oct_btn_shrink_styles' ) == 'oct_btn_shrink_none' ) ? '' : get_option( 'oct_btn_shrink_styles' );
-	$oct_btn_shadow = ( get_option( 'oct_btn_shadow_styles' ) == 'oct_btn_shadow_none' ) ? '' : get_option( 'oct_btn_shadow_styles' );
-	$oct_btn_slide = ( get_option( 'oct_btn_slide_styles' ) == 'oct_btn_slide_none' ) ? '' : get_option( 'oct_btn_slide_styles' );
-	echo $oct_btn_grow;
-	echo $oct_btn_shrink;
-	echo $oct_btn_shadow;
-	echo $oct_btn_slide;
-    ?>
-    	<script type="text/javascript">
-    		jQuery(document).ready(function($) { 
-				$( ".et_pb_button" ).addClass( "<?php echo $oct_btn_grow; ?>" );
-				$( ".et_pb_button" ).addClass( "<?php echo $oct_btn_shrink; ?>" );
-				$( ".et_pb_button" ).addClass( "<?php echo $oct_btn_shadow; ?>" );
-				$( ".et_pb_button" ).addClass( "<?php echo $oct_btn_slide; ?>" );
-			});
-    	</script>
-    <?php
+/**
+ * Customizer Javascript & CSS Output
+ *
+ * @action wp_footer
+ */
+function oct_customize_output() {
+	$grow_class   = get_option( 'oct_btn_grow_styles' );
+	$shrink_class = get_option( 'oct_btn_shrink_styles' );
+	$shadow_class = get_option( 'oct_btn_shadow_styles' );
+	$slide_class  = get_option( 'oct_btn_slide_styles' );
+
+	if ( 'oct_btn_grow_none' === $grow_class ) {
+		$grow_class = '';
+	}
+
+	if ( 'oct_btn_shrink_none' === $shrink_class ) {
+		$shrink_class = '';
+	}
+
+	if ( 'oct_btn_shadow_styles' === $shadow_class ) {
+		$shadow_class = '';
+	}
+
+	if ( 'oct_btn_slide_styles' === $slide_class ) {
+		$slide_class = '';
+	}
+
+	$button_classes = implode( ' ', array( $grow_class, $shrink_class, $shadow_class, $slide_class ) );
```

Comment:
By doing this (concatenating all the classes with a space between each one), we can cut our jQuery down to just one line.